server/services/reportService.test.ts

```diff
@@ -1,28 +1,21 @@
-const moment = require('moment')
-const serviceCreator = require('./reportService')
-const { ReportStatus } = require('../config/types')
-
-const incidentClient = {
-  getCurrentDraftReport: jest.fn(),
-  getReport: jest.fn(),
-  getReports: jest.fn(),
-  getReportForReviewer: jest.fn(),
-  getReportsForReviewer: jest.fn(),
-  updateDraftReport: jest.fn(),
-  createDraftReport: jest.fn(),
-  submitReport: jest.fn(),
-  deleteReport: jest.fn(),
-  commitAndStartNewTransaction: jest.fn(),
-  getIncompleteReportsForReviewer: jest.fn(),
-  getCompletedReportsForReviewer: jest.fn(),
-  updateAgencyId: jest.fn(),
-}
+import moment from 'moment'
+import IncidentClient from '../data/incidentClient'
+import ReportService from './reportService'
+import { OffenderService } from '../types/uof'
 
-const client = 'client-1'
+jest.mock('../data/incidentClient')
 
-const db = {
-  inTransaction: fn => fn(client),
-}
+const incidentClient = new IncidentClient(jest.fn as any, jest.fn() as any) as jest.Mocked<IncidentClient>
+
+const offenderService = {
+  getOffenderNames: jest.fn(),
+  getOffenderImage: jest.fn(),
+  getOffenderDetails: jest.fn(),
+  getPrisonersDetails: jest.fn(),
+} as jest.Mocked<OffenderService>
```

Comment:
can't generate a mock class for this as OffenderService is an interface. Do get compilation failure if it doesn't match the shape of the service though which is nice